keyboards/halfcliff/matrix.c

```diff
@@ -0,0 +1,277 @@
+/*
```

Comment:
What you want to do is the same as a keyboard such as angel64, using COL2ROW and ROW2COL together.
It is described so that this can be done only by changing the matrix_scan function.
Other functions are the same as standard.

At first I was considering "CUSTOM_MATRIX = lite" because I only changed the matrix_scan function.
But this didn't work with split keyboards.
Therefore, it is not used this time.
I think the reason is that when using the matrix_scan_custom function, the argument for the split keyboard cannot be used in the debounce function.

Reply:
> using COL2ROW and ROW2COL together

Ah, that's a good reason, okay.  

And sorry, I didn't look in depth at the code.  

And you're right, the lite custom matrix doesn't work with splits right now... Didn't realize that.

Reply:
I don't mind it :)
Eventually, I would like to make some contribution to QMK firmware.